types/utils_test.go

```diff
@@ -1,276 +1,18 @@
-			procCommandFile: initCommandFile,
-			tmpPerUserDir:   dirForOwner,
-			result:          dirForOwner,
-		},
-		{
-			homeRuntime:     homeRuntimeDisabled,
-			procCommandFile: initCommandFile,
-			tmpPerUserDir:   dirToBeCreated,
-			result:          dirToBeCreated,
-		},
-		{
-			homeRuntime:     homeRuntimeDisabled,
-			procCommandFile: initCommandFile,
-			tmpPerUserDir:   "", // Accessing tmp per user dir fails
-			homeDir:         dirForOwner,
-			result:          filepath.Join(dirForOwner, "rundir"),
-		},
-		{
-			homeRuntime:     homeRuntimeDisabled,
-			procCommandFile: initCommandFile,
-			tmpPerUserDir:   dirForAll,
-			homeDir:         dirForOwner,
-			result:          filepath.Join(dirForOwner, "rundir"),
-		},
-	}
-
-	for i, env := range envs {
-		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
-			os.Remove(dirToBeCreated)
-
-			resultDir, err := getRootlessRuntimeDirIsolated(env)
-			assert.NilError(t, err)
-			assert.Assert(t, resultDir == env.result)
-		})
-	}
-}
-
-type rootlessRuntimeDirEnvironmentRace struct {
-	procCommandFile string
-	tmpPerUserDir   string
-}
-
-func (env rootlessRuntimeDirEnvironmentRace) getProcCommandFile() string {
-	return env.procCommandFile
-}
-
-func (rootlessRuntimeDirEnvironmentRace) getRunUserDir() string {
-	return ""
-}
-
-func (env rootlessRuntimeDirEnvironmentRace) getTmpPerUserDir() string {
-	return env.tmpPerUserDir
-}
-
-func (rootlessRuntimeDirEnvironmentRace) homeDirGetRuntimeDir() (string, error) {
-	return "", errors.New("homedirGetRuntimeDir is disabled")
-}
-
-func (env rootlessRuntimeDirEnvironmentRace) systemLstat(path string) (*system.StatT, error) {
-	if path == env.tmpPerUserDir {
-		st, err := system.Lstat(path)
-		// We can simulate that race directory was created immediately after system.Lstat call.
-		if err := os.Mkdir(path, 0o700); err != nil {
-			return nil, err
-		}
-		return st, err
-	}
-	return system.Lstat(path)
-}
-
-func (rootlessRuntimeDirEnvironmentRace) homedirGet() string {
-	return homedir.Get()
-}
-
-func TestRootlessRuntimeDirRace(t *testing.T) {
-	raceDir := t.TempDir()
-
-	procCommandFile := filepath.Join(raceDir, "command")
-	err := os.WriteFile(procCommandFile, []byte("init"), 0o644)
-	assert.NilError(t, err)
-
-	tmpPerUserDir := filepath.Join(raceDir, "tmp")
-
-	resultDir, err := getRootlessRuntimeDirIsolated(rootlessRuntimeDirEnvironmentRace{
-		procCommandFile,
-		tmpPerUserDir,
-	})
-	assert.NilError(t, err)
-	assert.Assert(t, resultDir != tmpPerUserDir, "Rootless runtime dir shouldn't follow race dir.")
-}
-
 func TestDefaultStoreOpts(t *testing.T) {
-	storageOpts, err := defaultStoreOptionsIsolated(true, 1000, "./storage_test.conf")
-
-	expectedPath := filepath.Join(os.Getenv("HOME"), "1000", "containers/storage")
+	storageOpts, err := loadStoreOptionsFromConfFile("./storage_test.conf")
+	expectedPath := filepath.Join(os.Getenv("HOME"), fmt.Sprintf("%d", unshare.GetRootlessUID()), "containers/storage")
```

Comment:
[1] `…Isolated` had the `(rootless, UID)` parameters so that this test succeeds in all environments, even when run as root.

That would no longer be the case with these changes, but I also don’t know how much that matters. Maybe an explicit test skip would be enough. (Ideally I’d like the config logic to have good unit test coverage, but that’s clearly not this PR.)